lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -94,6 +94,23 @@ cl::opt<bool> SPIRVAllowUnknownIntrinsics(
     cl::desc("Unknown LLVM intrinsics will be translated as external function "
              "calls in SPIR-V"));
 
+enum FPContract {
+  FPContractOn,
+  FPContractOff,
+  FPContractFast,
+};
+
+cl::opt<FPContract> FPContractMode(
+    "ffp-contract", cl::desc("Fused FP operations:"), cl::init(FPContractOn),
+    cl::values(
+        clEnumValN(FPContractOn, "on",
+                   "choose a mode according to presence of fused LLVM "
+                   "intrinsics"),
+        clEnumValN(FPContractOff, "off", "disable for all entry points"),
+        clEnumValN(
+            FPContractFast, "fast",
+            "allow all operations to be contracted for all entry points")));
```

Comment:
Could you please refactor this option, so it can be used via library interface too? See d9063e9b770566646810fc72e3d1ed114c60eb39 and 5e01f2cf4ce54786115cbdf90bcfa9e169322cc1 as examples

Reply:
Done.